### DISCUSSION TITLE
Notification Banner

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#281] [Component suggestions]
<img width="480" alt="visuele weergave van een notification banner component" src="https://github.com/user-attachments/assets/92658a29-7dc8-4283-8e02-6f6ad9d18597">

We willen graag meer leren over de Notification Banner component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, GitHub of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack]({url-slack-bericht})

## Naam
Notification Banner

## Doel
Toont een informerend bericht dat niet direct is gerelateerd aan de inhoud van een pagina.

## Ook bekend als
Alert Banner, Banner

## Gerelateerde componenten
[Alert](https://github.com/orgs/nl-design-system/discussions/301), [Note](https://github.com/orgs/nl-design-system/discussions/407)

## Onderzoek

[Notification banner - GOV.UK](https://design-system.service.gov.uk/components/notification-banner/).
Er wordt onder andere beschreven dat:
- Men de notificatie kan missen als deze te vaak wordt ingezet.
- Dat de notificatie direct boven de H1 gepositioneerd moet worden.

Onder 'Research on this component' wordt daarnaast beschreven dat ze graag willen achterhalen:
- Hoe vaak het voorkomt dat gebruikers belangrijke informatie missen die in de Notification Banners staat. Dat geldt ook voor gebruikers die gebruik maken van ondersteunende technologie die wellicht direct naar de h1 navigeren.
- Of het helpt om de notificatie te sluiten, en hoe dat dan zou moeten gebeuren.

Er is ook een [link naar de Notification Banner Github discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/2) waar meer informatie onderzoek en aanpassingen worden gedeeld.

**Comment by jeffreylauwers** [comment]
## Amsterdam

<img width="600" alt="image" src="https://github.com/user-attachments/assets/997a0596-9125-472a-b1a7-afccda710349">

Amsterdam heeft een component genaamd 'Alert'. Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11030115) vastgelegd.

In [de documentatie van de Alert](https://designsystem.amsterdam/?path=/docs/components-feedback-alert--docs#guidelines) wordt onder andere beschreven dat men de oranje Alert direct onder de Header zou moeten plaatsen voor belangrijke en urgente informatie. Als voorbeelden worden systeemuitval of wijzigingen in de openingstijden van een stadskantoor aangedragen.

**Comment by jeffreylauwers** [comment]
## Den Haag

<img width="640" alt="image" src="https://github.com/user-attachments/assets/80040dd6-3c02-4477-af5a-ce8c2be1ecc5">

Den Haag heeft een component genaamd 'Alert'. Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11030620) vastgelegd.

In [de documentatie van de Alert](https://nl-design-system.github.io/denhaag/?path=/docs/css-data-display-alert--docs) wordt onder andere beschreven dat:

- De component bovenaan de pagina of sectie worden geplaatst waarop ze van toepassing zijn
- De component onder de pagina- of sectie Heading worden geplaatst.
- De component gebruikt kan worden voor meldingen om gebruikers te informeren over belangrijke wijzigingen, actuele omstandigheden, updates of veranderingen in de systeemstatus.

In Figma (Huidig) is ook een ontwerp te vinden voor een 'Calamity Alert'.

![Scherm­afbeelding 2024-12-23 om 09 02 20](https://github.com/user-attachments/assets/30243e4c-c084-46ee-9abe-ac264db09527)

In de documentatie staat beschreven dat een Calamity Alert wordt gebruikt wanneer de aandacht van de gebruiker heel belangrijk is. En dat deze kan worden gebruikt in situaties waarin belangrijke diensten voor de burgers plotseling niet beschikbaar zijn. De positie van de component is net onder de hoofdnavigatie of Breadcrumbs wanneer deze zichtbaar zijn.

**Comment by jeffreylauwers** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/66.0.0/components/on-page-interactives/banner)
- [Storybook](https://storybook.dso-toolkit.nl/66.0.0/?path=/story/core-banner--danger&viewMode=story)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/banner)

---

### Screenshots / Varianten

**Storybook**

Danger
![Scherm­afbeelding 2024-12-19 om 11 49 33](https://github.com/user-attachments/assets/b1224c16-1ebb-4949-913c-1dc4cfa5b1e7)


Danger with heading
![Scherm­afbeelding 2024-12-19 om 11 49 42](https://github.com/user-attachments/assets/e6e5b104-0543-4a48-8fca-54d996e7a6ec)


Error
![Scherm­afbeelding 2024-12-19 om 11 49 59](https://github.com/user-attachments/assets/49b071be-2c58-4813-88b1-2f3a5b1b8392)


Info
![Scherm­afbeelding 2024-12-19 om 11 50 09](https://github.com/user-attachments/assets/b3ce129d-dc61-4c40-bbd1-b67f3ad7f742)


Info compact non removable
![Scherm­afbeelding 2024-12-19 om 11 50 18](https://github.com/user-attachments/assets/9ae0ac5a-13c8-4f49-bf88-e1fd74bf9e1b)


Info compact
![Scherm­afbeelding 2024-12-19 om 11 50 30](https://github.com/user-attachments/assets/677d9560-a1f9-437c-bc2e-282c43251d33)

Rich info
![Scherm­afbeelding 2024-12-19 om 11 50 39](https://github.com/user-attachments/assets/8e69a374-57be-461e-8e68-b024f2bcdb42)


Rich warning
![Scherm­afbeelding 2024-12-19 om 11 50 51](https://github.com/user-attachments/assets/17ab533f-0c12-4c55-bb8f-0f0f6d6e3d71)


Warning
![Scherm­afbeelding 2024-12-19 om 11 51 00](https://github.com/user-attachments/assets/31188fa0-73cd-4a72-8bc1-a8dc0454163d)



---

### Notities

- Onder de naam 'Banner'.
- Documentatie beschikbaar. Daar wordt onder andere beschreven dat:
  - Banners verschijnen helemaal bovenaan het scherm en verschuiven de inhoud eronder. Ze laten algemene meldingen zien over het product of systeem, maar niet over specifieke gebruikersacties.
  - Ze blijven staan totdat de gebruiker ze verwijdert, of als de status die de banner veroorzaakte is opgelost. Banners kunnen optioneel een button of link bevatten, bijvoorbeeld naar meer informatie. Er mag slechts één banner tegelijk te zien zijn.
  - Banners blijven staan totdat de gebruiker ze verwijdert of als de status, die de banner veroorzaakte, is opgelost. Banners hebben altijd een ARIA `role="alert"`, zodat een screenreader ze meteen voorleest wanneer ze verschijnen. Screenreaders en andere hulptechnologie starten vaak bij de `<h1>`, dus zonder `role="alert"` worden ze gemist.

**Comment by jeffreylauwers** [comment]
## DUO

![Scherm­afbeelding 2024-12-19 om 11 32 48](https://github.com/user-attachments/assets/92c90f8e-c409-4bf1-80bb-7cb0a7907866)

DUO heeft een component genaamd 'Notification'. Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11037789) vastgelegd.

In [de documentatie van de Notification component](https://uno.dfront.rijkscloud.nl/#/components/notification) wordt onder andere beschreven dat:

- Een notificatie kan betrekking hebben op de actie die de gebruiker op dat moment uitvoert. In dat geval dient de notificatie inline getoond te worden.
- Wanneer een notificatie betrekking heeft op de gehele DUO omgeving (algehele storing), dan wordt de notificatie onder de navigatie getoond. In dat geval plaats je de notificatie als eerste element boven de main container.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

![Scherm­afbeelding 2024-12-19 om 12 07 59](https://github.com/user-attachments/assets/f4af94e2-d8c9-4477-9fd0-ec2f7303aacb)

OpenGemeenten heeft een component genaamd 'Waarschuwingen' (Notification in code). Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11038786) vastgelegd.

In [de documentatie van de Notification component](https://uno.dfront.rijkscloud.nl/#/components/notification) wordt onder andere beschreven dat:

- De component gebruikt kan worden om bezoekers te waarschuwen voor of wijzen op een bepaalde situatie.
- Het in sommige gevallen kan zijn dat de situatie dermate kritiek is dat de melding op elke pagina van de website getoond moet worden. En dat men in dat geval gebruik kan maken van een Calamiteit component. Momenteel is dit component niet beschikbaar.

Toevoeging van Martijn:

![Scherm_afbeelding 2024-12-19 om 14 33 11](https://github.com/user-attachments/assets/96785a89-b90e-46d9-be11-d645c3d177e2)

> Hierbij een oud ontwerp van hoe het eruit ziet op mobiel. Dit is een melding die zo belangrijk is dat die op iedere pagina staat, deze kun je wel inklappen (x) maar nooit wegklikken (in tegenstelling tot een alert). Omdat het ook mogelijk is dat er meerdere meldingen onder elkaar komen te staan (“afspraak maken niet mogelijk” en “gemeentehuis staat in brand”) hebben we ervoor gekozen om als tekst “toon alle gesloten meldingen” voor de link te gebruiken om de melding weer te laten zien.

**Comment by jeffreylauwers** [comment]
## RVO

![Scherm­afbeelding 2024-12-19 om 12 46 24](https://github.com/user-attachments/assets/981ad3f5-0129-49b4-b1f9-151e8cd6386d)

RVO heeft een component genaamd 'Alert'. Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11050479) vastgelegd.

Er is op dit moment geen documentatie beschikbaar waaruit blijkt of dit component ook als Notification Banner ingezet kan worden.

**Comment by jeffreylauwers** [comment]
## Utrecht

![Scherm­afbeelding 2024-12-19 om 12 48 41](https://github.com/user-attachments/assets/1b61dca3-c396-4987-b1bc-5ccc2745b1d8)

Utrecht heeft een component genaamd 'Alert'. Deze is bij de [Alert GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-11072895) vastgelegd.

In [de documentatie van de Alert component](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-alert--docs) worden gerelateerde componenten uiteengezet. Er wordt onder andere beschreven dat:

- Alert zo snel mogelijk wordt aangekondigd.
- Notification banner ook snel wordt aangekondigd, omdat het één van eerste onderdelen van de pagina is als je de banner plaatst aan het begin van de page header. Je kunt de banner overslaan een skip link.

Er is echter geen implementatie van de Notification Banner component te vinden.

**Comment by jeffreylauwers** [comment]
## Zijn dit ook Notification Banners?

Op dit moment vraag ik mij af of berichten die zelfs boven de [Page Header](https://github.com/orgs/nl-design-system/discussions/283) staan ook Notification Banners zijn.

### [NL Design System](https://www.nldesignsystem.nl/)
![Scherm­afbeelding 2024-12-19 om 13 53 45](https://github.com/user-attachments/assets/7f439716-0f22-4155-9b89-3abec28c8ebc)

### [Valsplat](https://valsplat.nl/)
![Scherm­afbeelding 2024-12-19 om 13 51 28](https://github.com/user-attachments/assets/77f82b0e-28d1-4f13-9741-19bb631b0789)

### [React Native](https://reactnative.dev/)
![Scherm­afbeelding 2024-12-19 om 13 57 45](https://github.com/user-attachments/assets/21ac7c4f-116a-402e-8158-62c450dece83)

### [Docusaurus](https://docusaurus.io/)
![Scherm­afbeelding 2024-12-19 om 13 58 16](https://github.com/user-attachments/assets/c2b19951-fd17-4dfb-8dcc-5a00fb4468f9)

### [Weeronline](https://www.weeronline.nl/)
![Screenshot_20241222_081340_Chrome](https://github.com/user-attachments/assets/26aaea7c-4435-40ff-a204-e905a9693798)

**Comment by rianrietveld** [comment]
@jeffreylauwers 
Ter overweging:
In web-code-land heet dit een [statusbericht](https://www.w3.org/Translations/WCAG22-nl/#dfn-status-messages), in het Engels een status message.
Is het een idee om dat gelijk te trekken?

**Reply by rianrietveld** [reply]
Plan: op een Design Dag kijken naar de componenten [Alert](https://github.com/orgs/nl-design-system/discussions/301), [Alert Dialog](https://github.com/orgs/nl-design-system/discussions/302), [Invalid Form Alert](https://github.com/orgs/nl-design-system/discussions/278) en [Notification Banner](https://github.com/orgs/nl-design-system/discussions/281), onderzoeken hoe deze verschillen en waarin ze overeenkomen, om te zien of we kunnen samenvoegen of beter beschrijven qua gebruik maar ook qua werking.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Notification Banner op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=notification+banner)

**Comment by jeffreylauwers** [comment]
Tijdens [Candidate voorbereidingsfase: Versimpeld of gesplitst voor Alert](https://github.com/orgs/nl-design-system/discussions/301#discussioncomment-17072777) is besloten dat er geen sluit-functie wordt toegevoegd aan de Alert. Maar dat deze functionaliteit wel worden toegevoegd aan de Notification Banner als deze richtig Candidate gaat.

**Comment by RenateRoke** [comment]
Voorbeeld Gemeente Midden-Groningen

<img width="2939" height="155" alt="www midden-groningen nl_" src="https://github.com/user-attachments/assets/eef61e8c-51f4-4ede-b7e3-2b7f06dfad85" />


```
<section
  id="block-toptasks-sub-theme-dvg-notification"
  class="dvg-notification dvg-notification--banner"
  data-once="dvgNotificationClose"
>
  <div
    class="dvg-notification--notification dvg-notification__wrapper dvg-notification__wrapper--notification"
  >
    <div
      class="dvg-notification__wrapper__inner dvg-notification__wrapper__inner--notification"
    >
      <svg
        class="dvg-notification__icon dvg-notification--banner__icon dvg-notification__icon--notification"
        aria-hidden="true"
        focusable="false"
        role="img"
      >
        <use
          xlink:href="/themes/contrib/toptasks_base_theme/dist/images/sprite.svg#icon-notification--notification"
        ></use>
      </svg>
      <h2
        class="dvg-notification__title dvg-notification--banner__title visually-hidden"
      >
        Belangrijke notificatie
      </h2>

      <div class="text-long">
        <p>
          Vanwege de verwachte hitte gelden op woensdag 29 juli
          <a href="https://www.midden-groningen.nl/afval" target="_blank"
            >aangepaste tijden voor de afvalinzameling en het afvalbrengpunt</a
          >.
        </p>
      </div>
      <button
        title="Sluit deze notificatie"
        aria-labelledby="dvg-notification__close-text-hidden"
        class="dvg-notification__close dvg-notification--banner__close"
        data-hash="xTReTMZ566rv3-Q0HGf3xg"
      >
        <svg
          class="dvg-notification__close-icon"
          width="30"
          height="30"
          aria-hidden="true"
          focusable="false"
          role="img"
        >
          <use
            xlink:href="/themes/contrib/toptasks_base_theme/dist/images/sprite.svg#icon-close"
          ></use>
        </svg>
        <span class="dvg-notification__close-text" aria-hidden="true"
          >Sluiten</span
        >
        <span
          id="dvg-notification__close-text-hidden"
          class="dvg-notification__close-text-hidden visually-hidden"
          >Sluit deze notificatie</span
        >
      </button>
    </div>
  </div>
</section>
```

**Comment by RenateRoke** [comment]
Voorbeeld Gemeente Alkmaar

<img width="2939" height="293" alt="www alkmaar nl_" src="https://github.com/user-attachments/assets/dd0622f3-1595-47ec-9a65-555215491807" />


```
<div
  id="js-emergency-notification"
  class="alignfull bg-subtle py-8 lg:py-12"
  style="anchor-name: --emergency-notification"
>
  <div
    class="container grid items-start gap-x-10 gap-y-6 md:grid-cols-[auto_1fr] lg:grid-cols-[auto_1fr_auto]"
  >
    <i class="fa-thin fa-bell text-secondary text-8" aria-hidden="true"></i>

    <div class="*:max-w-4xl">
      <h2 class="text-h3 mb-3">Welkom op onze vernieuwde website</h2>

      <div class="*:last:mb-0">
        <p style="font-weight: 400">
          U bent op de officiële website van&nbsp;gemeente Alkmaar. De website
          heeft een&nbsp;nieuw&nbsp;uiterlijk gekregen en is
          gebruiksvriendelijker, ook op mobiele apparaten.
        </p>
      </div>
    </div>

    <div class="grid justify-start gap-y-4 md:max-lg:col-start-2">
      <button
        class="is-button is-button-outline"
        id="js-emergency-notification-close-button"
        data-cookie="emergency-notification-end-date-1790719200"
      >
        <span>Melding sluiten</span>
        <i class="fa-regular fa-times" aria-hidden="true"></i>
      </button>
    </div>
  </div>
</div>
```

**Comment by RenateRoke** [comment]
Voorbeeld Gemeente Drimmelen

<img width="1944" height="290" alt="drimmelen nl_" src="https://github.com/user-attachments/assets/6c1a1580-e386-49bf-82ef-4d6dd508d74c" />


```
<div
  id="block-drimmelen-primaryactionbannerblock"
  class="m-banner js-banner--primary"
>
  <div class="t-layout t-layout--theme-banner m-banner--is-information">
    <div class="t-container">
      <div class="t-grid">
        <div class="t-grid__col">
          <h2 class="m-banner__title">
            Let op: vervroegde afvalinzameling en gewijzigde openingstijden
            Milieustraat
          </h2>
          <div class="m-banner__content">
            <p>
              Vanwege de hitte start de afvalinzameling deze week eerder op de
              dag op woensdag, donderdag en vrijdag. Zet uw afval op de
              inzameldag vóór 07.00 uur aan de straat.&nbsp;<br />GEWIJZIGDE
              openingstijden Milieustraat op woensdag 29 juli:
              <strong>9.30 uur tot 12.30 uur!</strong>
            </p>
          </div>
        </div>
        <div class="t-grid__col t-grid__col--shrink">
          <button
            class="m-banner__link u-spacing-pt--collapse u-spacing-pb--collapse js-banner"
            aria-label="Sluit de banner"
          >
            <span class="m-icon" aria-hidden="true">
              <svg
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
              >
                <path fill="none" d="M0 0h24v24H0V0z"></path>
                <path
                  d="M18.3 5.71c-.39-.39-1.02-.39-1.41 0L12 10.59 7.11 5.7c-.39-.39-1.02-.39-1.41 0-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
                ></path>
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
```
